File: core/src/main/java/forpleuvoir/dhwu1a/core/config/Dhwu1aConfig.java
package forpleuvoir.dhwu1a.core.config;

import com.google.gson.annotations.SerializedName;
import forpleuvoir.dhwu1a.core.util.JsonUtil;

/**
 * 配置
 *
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.config
 * <p>#class_name Dhwu1aConfig
 * <p>#create_time 2021/6/28 20:34
 */
public class Dhwu1aConfig {
    @SerializedName("botId")
    public final Long botId;
    @SerializedName("ip")
    public final String ip;
    @SerializedName("port")
    public final Integer port;
    @SerializedName("verifyKey")
    public final String verifyKey;
    @SerializedName("logConfig")
    public LogConfig logConfig;

    public Dhwu1aConfig(Long botId, String ip, Integer port, String verifyKey, LogConfig logConfig) {
        this.botId = botId;
        this.ip = ip;
        this.port = port;
        this.verifyKey = verifyKey;
        this.logConfig = logConfig;
    }

    @Override
    public String toString() {
        return JsonUtil.toJsonStr(this);
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/config/Dhwu1aConfig.kt
package forpleuvoir.dhwu1a.core.config

import com.google.gson.JsonElement
import com.google.gson.annotations.SerializedName
import forpleuvoir.dhwu1a.core.common.IJsonData

/**
 * 配置类
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.config
 *
 * #class_name Dhwu1aConfig
 *
 * #create_time 2021/8/9 23:34
 */
class Dhwu1aConfig(botId: Long, ip: String, port: Int, verifyKey: String, logConfig: LogConfig) : IJsonData, IConfig {

    @SerializedName("botId")
    var botId: Long = botId
        private set

    @SerializedName("ip")
    var ip: String = ip
        private set

    @SerializedName("port")
    var port: Int = port
        private set

    @SerializedName("verifyKey")
    var verifyKey: String = verifyKey
        private set

    @SerializedName("logConfig")
    var logConfig: LogConfig = logConfig
        private set


    override fun fromJson(jsonElement: JsonElement): Dhwu1aConfig {
        if (jsonElement.isJsonObject) {
            val json = jsonElement.asJsonObject
            botId = json.get("botId").asLong
            ip = json.get("ip").asString
            port = json.get("port").asInt
            verifyKey = json.get("verifyKey").asString
            logConfig = LogConfig().fromJson(json.get("logConfig"))
        }
        return this
    }

}
